zerver/views/users.py

```diff
@@ -119,6 +119,8 @@ def deactivate_user_backend(
     user_profile: UserProfile,
     *,
     user_id: PathOnly[int],
+    is_spammer: Json[bool] = True,
+    message_delete_action: Json[int] = 0,
```

Comment:
This should also use the constant and the default value for `is_spammer` should be `False`.

Reply:
Are we allowing deleting messages without marking the user as spammer? Here it is allowed but `do_deactivate_user` will not delete messages if `is_spammer` is not `True`.

Reply:
We can raise error if we do not want to allow. Can be discussed once on CZO.